src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/AccountAuthorizedDepositorsProcessor.cs

```diff
@@ -0,0 +1,390 @@
+    {
+        var rowsInserted = 0;
+
+        rowsInserted += await CopyAccountAuthorizedDepositorHistory();
+        rowsInserted += await CopyAccountAuthorizedDepositorAggregateHistory();
+
+        return rowsInserted;
+    }
+
+    private AccountAuthorizedDepositorsDbLookup GetDbLookup(
+        CoreModel.AccountAuthorizedDepositorEntrySubstate accountAuthorizedDepositorEntrySubstate,
+        AccountAuthorizedDepositorsChangePointerLookup lookup
+    )
+    {
+        AccountAuthorizedDepositorsDbLookup entryLookup;
+
+        switch (accountAuthorizedDepositorEntrySubstate.Key.Badge)
+        {
+            case CoreModel.ResourceAuthorizedDepositorBadge resourceBadge:
+            {
+                var resourceEntityId = _referencedEntityDictionary.Get((EntityAddress)resourceBadge.ResourceAddress).DatabaseId;
+
+                entryLookup = new AccountAuthorizedDepositorsDbLookup(
+                    AccountEntityId: lookup.AccountEntityId,
+                    ResourceEntityId: resourceEntityId,
+                    NonFungibleIdDataId: null
+                );
+                break;
+            }
+
+            case CoreModel.NonFungibleAuthorizedDepositorBadge nonFungibleBadge:
+            {
+                var globalIdDatabaseId = _referencedNonFungibleIdDictionary.Get(
+                    new NonFungibleGlobalIdLookup(
+                        (EntityAddress)nonFungibleBadge.NonFungibleGlobalId.ResourceAddress,
+                        nonFungibleBadge.NonFungibleGlobalId.LocalId.SimpleRep)
+                );
+
+                entryLookup = new AccountAuthorizedDepositorsDbLookup(
+                    AccountEntityId: lookup.AccountEntityId,
+                    ResourceEntityId: globalIdDatabaseId.ResourceEntityId,
+                    NonFungibleIdDataId: globalIdDatabaseId.NonFungibleIdDataId
+                );
+                break;
+            }
+
+            default:
+                throw new UnreachableException(
+                    $"Expected either ResourceAuthorizedDepositorBadge or NonFungibleGlobalAuthorizedDepositorBadge but found {accountAuthorizedDepositorEntrySubstate.Key.Badge.GetType()}");
+        }
+
+        return entryLookup;
+    }
+
+    private async Task<IDictionary<AccountAuthorizedDepositorsDbLookup, AccountAuthorizedDepositorHistory>> MostRecentAccountAuthorizedDepositorHistory()
+    {
+        var lookupSet = new HashSet<AccountAuthorizedDepositorsDbLookup>();
+
+        foreach (var (lookup, change) in _accountAuthorizedDepositorsChanges.AsEnumerable())
+        {
+            foreach (var entry in change.AuthorizedDepositorEntries)
+            {
+                lookupSet.Add(GetDbLookup(entry, lookup));
+            }
+        }
+
+        var entityIds = lookupSet.Select(x => x.AccountEntityId).ToList();
+        var resourceEntityIds = lookupSet.Select(x => x.ResourceEntityId).ToList();
+        var nonFungibleIdDataIds = lookupSet.Select(x => x.NonFungibleIdDataId).ToList();
```

Comment:
Hint: there's this `unzip` extension method we could use over here, but this is not a big deal.